loader/src/sources/riteaid/common.js

```diff
@@ -9,4 +9,46 @@ class RiteAidApiError extends HttpApiError {
   }
 }
 
-module.exports = { RiteAidApiError };
+/**
+ * Get the external IDs for a given Rite Aid store number. Rite Aid has a
+ * number of sub-brands, and sometimes we need external IDs for both the whole
+ * Rite Aid system and for the store's sub-brand. Luckily, the Rite-Aid wide ID
+ * is always "<sub-brand-specific prefix(es)><sub-brand ID>", so we can
+ * determine the sub-brand and sub-brand ID from the Rite Aid ID.
+ * (Per discussion in 2021 with Rite Aid tech team.)
+ *
+ * For example, the Rite Aid ID "06958" is also the Bartell ID "58".
+ *
+ * @param {number|string} storeNumber The Rite Aid store number.
+ * @returns {[string, string][]}
+ */
+function getExternalIds(storeNumber) {
+  const numberString = storeNumber.toString();
+  const result = [["rite_aid", numberString]];
+  if (/69\d\d$/.test(numberString)) {
+    result.push(["bartell", numberString.slice(-2)]);
+  }
+  return result;
+}
+
+/**
+ * Get a human-friendly name for a Rite Aid location. If the store is part of a
+ * known sub-brand, this uses the sub-brand's name instead of "Rite Aid".
+ *
+ * @param {[string, string][]} externalIds The store's external IDs
+ * @returns {string}
+ */
+function getLocationName(externalIds) {
+  let riteAidId;
+  for (const [system, value] of externalIds) {
+    if (system === "rite_aid") {
+      riteAidId = value;
+    } else if (system === "bartell") {
+      return `Bartell Drugs #${value}`;
+    }
+  }
+
+  return `Rite Aid #${riteAidId}`;
```

Comment:
Maybe make returning this conditional on whether `riteAidId` got set to a useful value?

Alternately, could return immediately on finding a `rideAidId` value.

Reply:
Ah, yeah, fair point. This is outside the loop because I was trying to be sensitive to the fact that the Rite Aid ID and sub-brand ID could come in any order, and we want to prefer the sub-brand. (In practical terms, this is always coming from the `getExternalIds()` function, so we could just iterate in reverse order, and then the first one would always be the more specific one.

Reply:
Rejiggered this a bit to cover not finding a `rite_aid` system, and maybe read a little bit more clearly about why it doesn’t return immediately.